### DISCUSSION TITLE
Issue with inference running - Error when reading video frame. Stopping video reader.

### DISCUSSION THREAD

**Discussion by GettyScience2** [talmolab/sleap-nn#658] [Q&A] [answered]
Hello, 

I have successfully made a bottom-up model that appears to be working well in tracking but is failing to during inference running. I have four videos that the model was trained on and am attempting to run the inference on them to compare to a DeepLabCut output. 

Below is my input for the model being used and the inference/tracking method I am attempting:

<img width="1262" height="992" alt="image" src="https://github.com/user-attachments/assets/279e183c-b53a-4771-9dbd-b497ac012f0f" />

<img width="1202" height="900" alt="image" src="https://github.com/user-attachments/assets/e0d5b492-67ea-493b-a3d5-0b3b20da5ab6" />

I am working with 10 minute videos of two fish interacting. Camera angle is in front of the tank currently. Occlusions are likely, which is why I have chosen the bottom-up model and the flow method for tracking. 

When I attempt to run inference on the first video, I get this output:

Running: sleap track --gui --data_path C:/Users/giraf/Desktop/sleap_trial/labels.v003.slp --video_index 0 --frames 0,-18084 --model_paths C:\Users\giraf\Desktop\sleap_trial\models\New_Skeleton_2.bottomup.n=195 -o C:/Users/giraf/Desktop/sleap_trial\predictions\pair1.mp4_labels.v003.slp.260624_121513.predictions.slp --max_instances 2 --tracking --track_matching_method greedy --tracking_window_size 5 --candidates_method local_queues --max_tracks 4 --use_flow --features keypoints --scoring_method oks
W0624 12:15:16.694000 2268 Lib\site-packages\torch\utils\flop_counter.py:29] triton not found; flop counting will not work for triton kernels
2026-06-24 12:15:20 | Started inference at: 2026-06-24 12:15:20.359150
2026-06-24 12:15:20 | sleap-nn 0.2.0 | Python 3.13.14 | PyTorch 2.12.0+cu130 | CUDA 13.0 | 1 GPU(s)
2026-06-24 12:15:20 | Using device: cuda
2026-06-24 12:15:26 | Error when reading video frame. Stopping video reader.
Failed to read frame index 131.
2026-06-24 12:15:41 | Finished inference at: 2026-06-24 12:15:41.840926
2026-06-24 12:15:41 | Total runtime: 21.481783866882324 secs
2026-06-24 12:15:43 | Predictions output path: C:/Users/giraf/Desktop/sleap_trial\predictions\pair1.mp4_labels.v003.slp.260624_121513.predictions.slp
2026-06-24 12:15:43 | Saved file at: 2026-06-24 12:15:43.454038
[h264 @ 0000020416672c40] negative number of zero coeffs at 22 49
[h264 @ 0000020416672c40] error while decoding MB 22 49
[h264 @ 0000020416673100] Invalid NAL unit size (33554435 > 73463).
[h264 @ 0000020416673100] Error splitting the input into NAL units.
[h264 @ 0000020416673a00] negative number of zero coeffs at 65 18
[h264 @ 0000020416673a00] error while decoding MB 65 18
[h264 @ 0000020416673580] Invalid NAL unit size (536871426 > 69718).
[h264 @ 0000020416673580] Error splitting the input into NAL units.
[h264 @ 000002041fe3f380] negative number of zero coeffs at 22 49
[h264 @ 000002041fe3f380] error while decoding MB 22 49
[h264 @ 000002041fe3c5c0] Invalid NAL unit size (33554435 > 73463).
[h264 @ 000002041fe3c5c0] Error splitting the input into NAL units.
[h264 @ 000002041fe405c0] negative number of zero coeffs at 65 18
[h264 @ 000002041fe405c0] error while decoding MB 65 18
[h264 @ 000002041fe40a40] Invalid NAL unit size (536871426 > 69718).
[h264 @ 000002041fe40a40] Error splitting the input into NAL units.

I get the same failing point whether I use the simple or the flow tracking method. When I run inference on all four of my videos, I get a a later failing frame for video 2 and even later with video 3 while video 4 has no errors. 

Any suggestions?

**Comment by tom21100227** [comment]
Hi @GettyScience2, see my reply on your original issue (https://github.com/talmolab/sleap/issues/2776). 

I beleive both comes from a slightly currupted / unconventional video structure. So the reencoding step should definitely help. To ensure the model is trained on the correct frames, retrain the model with the reencoded videos before you run any inference on it! 

In case the fix did not work, we would need more information to better help resolve your issue and make sure other users don't experience the same friction you're experiencing when using sleap. if you can provide us some files to help us recreate the issue it would help a lot: 
- the .slp file and/or,
- a sample videos of yours. 
-  output of `sleap doctor` on your computer. (depending on how you use sleap, it might be `uv run sleap doctor`

Google form link here: https://docs.google.com/forms/d/e/1FAIpQLSfWBwcuZH8lxGyFabMAODLjSUE4YrSCYbrrnYvXCpgBqZpcJA/viewform

Best,
Tom

**Comment by GettyScience2** [comment, marked as answer]
Hello! I am sorry for the late response and (basically) double post. I was successfully able to reconfigure the video and perform inference running based on a previous suggestion in https://github.com/talmolab/sleap/issues/2688:

ffmpeg -i your_video.mp4 -c:v libx264 -preset medium -crf 18 reencoded_video.mp4

Noticing your suggestion to retrain here, however, I will have to do that next. Regardless, it looks like the reencoding step is vital for these videos. 

Using the suggestion in that post led me to make a new version of the original video. Would reencoding the slp file itself lead to all of the original videos being encoded? If it works best to reencode the videos individually, should I give them the same name as the original videos to replace them?

Thank you!

**Reply by tom21100227** [reply]
Hi @GettyScience2 , 

Glad to hear it's working! sio reencode does the same thing that ffmpeg line does under the hood, it just automates the process and replace all videos in your slp files with the reencoded version. so if it works now i wouldn't worry about running reencoding again. 

Best,
Tom
